lib/active_force/sobject.rb

```diff
@@ -108,6 +112,20 @@ def create!
       self
     end
 
+    def uninitialize_attributes(attrs)
+        return if attrs.blank?
+        attr_set = @attributes.map do |key, value|
+          if attrs.include?(self.mappings.dig(key.name.to_sym))
+            ActiveModel::Attribute::UninitializedSobject.new(key.name, key.type)
+          else
+            key
+          end
+        end
+      attrs = attr_set.instance_variable_get(:@attributes) || {}
+      new_attr_set = ActiveModel::LazyAttributeSet.new({}, {}, {}, {}, attrs)
+      self.instance_variable_set(:@attributes, new_attr_set)
+    end
```

Comment:
Not loving this dance of replacing the all the attributes with the `LazyAttributeSet` class.  Just curious... do you actually need that or can you use the simpler `ActiveModel::AttributeSet` instance that we already have?  Looking through the code there doesn't seem to be much special about the `ActiveModel::LazyAttributeSet` class for this specific use case.  Could you do something like this?

```
def set_uninitialized_attributes(attrs)
  return if attrs.blank?
  attrs.each do |attr|
    attributes[attr] = ActiveModel::Attribute::UninitializedSobject.new(attributes[attr].name, attributes[attr].type)
  end
end
```

Reply:
Happy to pair on this... I _think_ I had something working in the REPL.

Reply:
Good call. I was able to mutate the attributes by accessing the @attributes instance variable inside of the @attributes instance variable. See code changes